v-next/hardhat-ethers/src/internal/signers/signers.ts

```diff
@@ -0,0 +1,280 @@
+import type { HardhatEthersProvider } from "../hardhat-ethers-provider/hardhat-ethers-provider.js";
+import type { NetworkConfig } from "@ignored/hardhat-vnext/types/config";
+import type { BlockTag, TransactionRequest, ethers } from "ethers";
+
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import {
+  assertArgument,
+  getAddress,
+  hexlify,
+  resolveAddress,
+  toUtf8Bytes,
+  TypedDataEncoder,
+} from "ethers";
+
+import { getRpcTransaction } from "../ethers-utils/ethers-utils.js";
+
+import { deepCopy } from "./deep-copy.js";
+import { populate } from "./populate.js";
+
+export class HardhatEthersSigner implements ethers.Signer {
+  readonly #gasLimit: bigint | undefined;
+
+  public readonly address: string;
+  public readonly provider: ethers.JsonRpcProvider | HardhatEthersProvider;
+  public static networkName: string;
+  public static networkConfig: NetworkConfig;
+
+  public static async create(
+    provider: HardhatEthersProvider,
+    networkName: string,
+    networkConfig: NetworkConfig,
+    address: string,
+  ): Promise<HardhatEthersSigner> {
+    this.networkName = networkName;
+    this.networkConfig = networkConfig;
+
+    // depending on the config, we set a fixed gas limit for all transactions
+    let gasLimit: bigint | undefined;
+
+    if (networkName === "hardhat") {
+      // If we are connected to the in-process hardhat network and the config
+      // has a fixed number as the gas config, we use that.
+      // Hardhat core already sets this value to the block gas limit when the
+      // user doesn't specify a number.
+      if (networkConfig.gas !== "auto") {
+        gasLimit = networkConfig.gas;
+      }
+    } else if (networkName === "localhost") {
+      const configuredGasLimit = networkConfig.gas;
+
+      if (configuredGasLimit !== "auto") {
+        // if the resolved gas config is a number, we use that
+        gasLimit = configuredGasLimit;
+      } else {
+        // if the resolved gas config is "auto", we need to check that
+        // the user config is undefined, because that's the default value;
+        // otherwise explicitly setting the gas to "auto" would have no effect
+        if (networkConfig.gas === undefined) {
+          // finally, we check if we are connected to a hardhat network
+          let isHardhatNetwork = false;
+          try {
+            await provider.send("hardhat_metadata");
+            isHardhatNetwork = true;
+          } catch {}
+
+          if (isHardhatNetwork) {
+            // WARNING: this assumes that the hardhat node is being run in the
+            // same project which might be wrong
+            // TODO: enable when V3 is ready: blockGasLimit currently missing in networkConfig
+            // gasLimit = networkConfig.blockGasLimit;
```

Comment:
As mentioned in the PR description, this is the property (blockGasLimit) that is currently not supported in `networkConfig`